src/WallpaperEngine/Render/CWallpaper.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <assert.h>
+
+#include "WallpaperEngine/Core/CWallpaper.h"
+
+namespace WallpaperEngine::Render
+{
+    class CWallpaper
+    {
+    public:
+        template<class T> const T* as () const { assert (is<T> ()); return (const T*) this; }
+        template<class T> T* as () { assert (is<T> ()); return (T*) this; }
+
+        template<class T> bool is () { return this->m_type == T::Type; }
+
+        CWallpaper (Core::CWallpaper* wallpaperData, std::string type);
+
+        Core::CWallpaper* getWallpaperData ();
```

Comment:
This function should be private and CScene and CVideo provide their own public version, that performs the required cast to ease the usage of the wallpaper's data on the code, for example:
`Core::CScene* getScene () { return this->getWallpaperData ()->as<Core::CScene> (); }`